Webpacker clean Sprockets and update docs

ferblape opened the pull request:
#### :tophat: What? Why?

This PR fixes pending issues after #7733 

- docs should explain about `decidim:webpacker:install`
- sprockets dependency was already present in some gems

#### :pushpin: Related Issues

- After #7733

#### Testing

Not code added here, I guess the gems should build succesfully 

#### :clipboard: Checklist
:rotating_light: Please review the [guidelines for contributing](https://github.com/decidim/decidim/blob/develop/CONTRIBUTING.adoc) to this repository.

- [ ] :question: **CONSIDER** adding a unit test if your PR resolves an issue.
- [ ] :heavy_check_mark: **DO** check open PR's to avoid duplicates.
- [ ] :heavy_check_mark: **DO** keep pull requests small so they can be easily reviewed.
- [ ] :heavy_check_mark: **DO** build locally before pushing.
- [ ] :heavy_check_mark: **DO** make sure tests pass.
- [ ] :heavy_check_mark: **DO** make sure any new changes are documented in `docs/`.
- [ ] :heavy_check_mark: **DO** add and modify seeds if necessary.
- [ ] :heavy_check_mark: **DO** add CHANGELOG upgrade notes if required.
- [ ] :heavy_check_mark: **DO** add to GraphQL API if there are new public fields.
- [ ] :heavy_check_mark: **DO** add link to MetaDecidim if it's a new feature.
- [ ] :x:**AVOID** breaking the continuous integration build.
- [ ] :x:**AVOID** making significant changes to the overall architecture.

### :camera: Screenshots
*Please add screenshots of the changes you're proposing*
![Description](URL)

:hearts: Thank you!

mrcasals commented:
@ahukkanen can you review this, please? 😄

ferblape commented:
The generators are failing again, I'll fix it

mrcasals commented:
@ahukkanen noteds! I ported your comment to #7984 so we can keep track of those, I'll merget his PR meanwhile!

mrcasals commented:
Actually @ferblape could you review those failing tests in Generators? Thanks!

ferblape commented:
I stil need to fix specs, I'll do ASAP

ferblape commented:
A quick update to this PR, I removed all sprockets dependencies from the gems and updated the description. Still blocked by https://github.com/decidim/decidim-bulletin-board/pull/185 which uses Sprockets and is breaking the specs

ahukkanen commented on decidim-comments/README.md:
```diff
@@ -47,11 +47,10 @@ For the backend side:
 
-- Typescript (introduced in #1001)
 - React
 - Apollo
```
By the way @ferblape comments no longer work with React/Apollo, so these can be also removed.

ferblape commented:
You are right, I'll fix it.

On Fri, 28 May 2021 at 09:39, Antti Hukkanen ***@***.***>
wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In decidim-comments/README.md
> <https://github.com/decidim/decidim/pull/7942#discussion_r641337441>:
>
> > @@ -47,11 +47,10 @@ For the backend side:
>
>  For the frontend side:
>
> -- Typescript (introduced in #1001)
>  - React
>  - Apollo
>
> By the way @ferblape <https://github.com/ferblape> comments no longer
> work with React/Apollo, so these can be also removed.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/decidim/decidim/pull/7942#pullrequestreview-671017414>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAAEJUHWGGUZD44SMH3V37DTP5CEFANCNFSM44FI72ZQ>
> .
>


-- 
Fernando Blat
***@***.***
+34 660825001

Populate / Tools for civic engagement
https://populate.tools

Project stories
twitter.com/populate_ & populate.tools/blog

ferblape commented:
@leio10 this is the never ending PR, after bboard updated there's still a dependency using Sprockets: graphiql-rails. 

The maintainer of the project is a bit tired of it and recommends stop using the gem and use some alternatives out there. Here's a proposal to replace the rails engine by a simple view, and require the assets as npm dependencies. I think this could fit in the decidim-api module. If you are ok with this I'll replace graphiql-rails with some code in the gem.

ferblape commented:
@leio10 at least, ready to review!

leio10 commented on docs/modules/develop/pages/guide_migrate_webpacker_app.adoc:
```diff
@@ -35,13 +35,24 @@ bundle install
+Steps it does:
+
+- copies babel and postcss configuration to instance application
+- copies config/webpacker.yml configuration to instance application
```
We should remove this after merging #8066, right?

ferblape replied:
Yes, fixed

leio10 commented on CHANGELOG.md:
```diff
@@ -4,6 +4,7 @@
 
 #### Webpacker migration
 As per [#7464](https://github.com/decidim/decidim/pull/7464), [#7733](https://github.com/decidim/decidim/pull/7733) Decidim has been upgraded to use Webpacker to manage its assets. It's a huge change that requires some updates in your applications. Please refere to the guide [Migrate to Webpacker an instance app](https://github.com/decidim/decidim/blob/develop/docs/modules/develop/pages/guide_mig ...
+### Notes
```
⚠️ Mmm, isn't this weird?

leio10 commented on decidim-core/lib/decidim/webpacker/webpack/custom.js:
```diff
@@ -99,6 +99,7 @@ module.exports = {
     decidim_assemblies: "DECIDIM_PATH/decidim-assemblies/app/packs/entrypoints/decidim_assemblies.js",
     decidim_assemblies_admin: "DECIDIM_PATH/decidim-assemblies/app/packs/entrypoints/decidim_assemblies_admin.js",
     decidim_api_docs: "DECIDIM_PATH/decidim-api/app/packs/entrypoints/decidim_api_docs.js",
+    decidim_api_graphiql: "DECIDIM_PATH/decidim-api/app/packs/entrypoints/decidim_api_graphiql.js",
```
⚠️ We should migrate this to the new way implemented in #8066

leio10 commented on decidim_app-design/config/webpack/custom.js:
```diff
@@ -99,6 +99,7 @@ module.exports = {
     decidim_assemblies: "../decidim-assemblies/app/packs/entrypoints/decidim_assemblies.js",
     decidim_assemblies_admin: "../decidim-assemblies/app/packs/entrypoints/decidim_assemblies_admin.js",
     decidim_api_docs: "../decidim-api/app/packs/entrypoints/decidim_api_docs.js",
+    decidim_api_graphiql: "../decidim-api/app/packs/entrypoints/decidim_api_graphiql.js",
```
⚠️ We should migrate this to the new way implemented in #8066

leio10 commented on decidim-generators/lib/decidim/generators/app_generator.rb:
```diff
@@ -239,7 +239,7 @@ def gem_modifier
         return if options[:path]
 
-        @branch ||= options[:edge] ? "develop" : options[:branch].presence
+        @branch ||= options[:edge] ? "webpacker-dep-docs" : options[:branch].presence
```
⚠️ Should we rollback this to develop before merging this?

ferblape replied:
If we rollback this before tests won't pass. I can create another PR just after this PR being merged

leio10 commented on decidim-generators/spec/generators_spec.rb:
```diff
@@ -109,7 +109,7 @@ module Decidim
 
       context "with --branch flag" do
-        let(:default_branch) { "develop" }
+        let(:default_branch) { "webpacker-dep-docs" }
```
⚠️ Should we rollback this to develop before merging this?

leio10 reviewed (changes requested):
Awesome and huge job! 👏👏💎
Let's resolve the issues that I've found and merge this!

ferblape commented:
@leio10 changes applied except the branch name. I can change the branch name and you merge it despite the generator specs start to fail?
